ui/lib/pki/addon/components/pki-key-parameters.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import type PkiRoleModel from 'vault/models/pki/role';
+import type PkiKeyModel from 'vault/models/pki/key';
+import type PkiActionModel from 'vault/models/pki/action';
+import type { HTMLElementEvent } from 'forms';
+/**
+ * @module PkiKeyParameters
+ * PkiKeyParameters components are used to display a list of key bit options depending on the selected key type. The key bits field is disabled until a key type is selected.
+ * If the component renders in a group, other attrs may be passed in and will be rendered using the <FormField> component
+ * @example
+ * ```js
+ * <PkiKeyParameters @model={{@model}} @fields={{fields}}/>
+ * ```
+ * @param {class} model - The pki/role, pki/action, pki/key model.
+ * @param {string} fields - Array of attributes from a formFieldGroup generated by the @withFormFields decorator ex: [{ name: 'attrName', type: 'string', options: {...} }]
+ */
+interface BitOptions {
```

Comment:
I don't think we've set a standard for this, but should we organize the order of types here? I think I'd expect to see this interface below, around line 39.

Reply:
Nice! I don't think we've decided. I always tend to group them based on how they're used, but alphabetical also makes sense!